doc/src/build-instructions.md

```diff
@@ -528,7 +528,23 @@ If you are building for a different target board the `CLOCK_FREQUENCY` definitio
 
 The rcc_clock_setup_XXX_XXX will also need to be changed to match your particular chip-set. Consult [ST's documentation](https://www.st.com/en/microcontrollers-microprocessors/stm32-32-bit-arm-cortex-mcus.html) for appropriate settings.
 
+### Configure logging with `cmake`
+The default maximum log level is `LOG_INFO`. To change the maximum level displayed pass `-DAVM_LOG_LEVEL_MAX="{level}"` to `cmake`, with one of `LOG_ERROR`, `LOG_WARN`, `LOG_INFO`, or `LOG_DEBUG` (listed from least to most verbose). Log messages can be completely disabled by using `-DAVM_LOG_DISABLE=on`.
+
+For log entries colorized by log level pass `-DAVM_ENABLE_LOG_COLOR=on` to cmake. With color enable there is a very small performance penalty (~1ms per message printed), the log entries are colored as follows:
+
+| Message Level |  Color  |
+|---------------|---------|
+| ERROR | Red |
+| WARN | Orange |
+| INFO | Green |
+| DEBUG | Blue |
+
+By default only `ERROR` messages contain file and line number information. This can be included with all log entries by passing `-DAVM_ENABLE_LOG_LINES=on` to cmake, but it does incur a significant performance penalty and is only suggested for debugging during development.
+
 ### Printing
+AtomVM is built with standard `newlib` to support `long long` integers (`signed` and `unsigned`). If you are building for a device with extremely limited flash space the `nano` version of `newlib` can be used instead. This may be done by passing `-DAVM_NEWLIB_NANO=on`. If the `nano newlib` is used logs be disabled. Many of the VM low level log messages will include `%ull` formatting and will cause buffer overflows and crash the VM if logging is not disabled for `nano newlib` builds. The total flash savings of using `nano newlib` and disabling logs is just under 40kB.
+
```

Comment:
“If the `nano_newlib` is used logs SHOULD? MUST? be disabled”. Some missing guidance here.

Reply:
Is there a way to force logging to be disabled in CMake if `AVM_NEWLIB_NANO=on`?

Reply:
Yes. In fact I changed the cmake rules to disable logs if newlib nano is used already. I just forgot to update the documentation.

Reply:
Fixed the missing "will" in documentation, and expanded for clarity.